bench/nested-deps/bench.mjs

```diff
@@ -165,35 +171,83 @@ function waitFor(millis) {
   return new Promise((resolve) => setTimeout(resolve, millis))
 }
 
-async function main() {
-  await fs.rmDir('.next', { recursive: true })
-  const file = new File('pages/index.jsx')
-  try {
+await fs.rm('.next', { recursive: true }).catch(() => {})
+const file = new File(join(CWD, 'pages/index.jsx'))
+const results = []
+
+try {
+  if (command === 'dev' || command === 'all') {
     const instance = await runNextCommandDev(['dev', '--port', '3000'])
-    const res = await fetch('http://localhost:3000/')
+
+    function waitForCompiled() {
+      return new Promise((resolve) => {
+        function waitForOnData(data) {
+          const message = data.toString()
+          const compiledRegex =
+            /compiled client and server successfully in (\d*[.]?\d+)\s*(m?s) \((\d+) modules\)/gm
+          const matched = compiledRegex.exec(message)
+          if (matched) {
+            resolve({
+              'time (ms)': (matched[2] === 's' ? 1000 : 1) * Number(matched[1]),
+              modules: Number(matched[3]),
+            })
+            instance.stdout.removeListener('data', waitForOnData)
+          }
+        }
+        instance.stdout.on('data', waitForOnData)
+      })
+    }
+
+    const [res, initial] = await Promise.all([
+      fetch('http://localhost:3000/'),
+      waitForCompiled(),
+    ])
     if (res.status !== 200) {
       throw new Error('Fetching / failed')
     }
 
-    await waitFor(3000)
+    results.push(initial)
 
     file.prepend('// First edit')
 
-    await waitFor(5000)
+    results.push(await waitForCompiled())
+
+    await waitFor(1000)
 
     file.prepend('// Second edit')
 
-    await waitFor(5000)
+    results.push(await waitForCompiled())
+
+    await waitFor(1000)
 
     file.prepend('// Third edit')
 
-    await waitFor(5000)
+    results.push(await waitForCompiled())
+
+    console.table(results)
 
     await killApp(instance)
-    await fs.rmDir('.next', { recursive: true })
-  } finally {
-    file.restore()
   }
+  if (command === 'build' || command === 'all') {
+    // ignore error
+    await fs.rm('.next', { recursive: true }).catch(() => {})
+
+    execSync(`node ${NEXT_BIN} build ./bench/nested-deps`, {
```

Comment:
curiousity q, other passes in here uses async version of interfaces but only this one uses `execSync`. is there specific reason?

Reply:
The `spawn` usage in the `next dev` command is for streaming stdout without blocking the following logic (modify files and triggering fast refresh). The `build` command is one-shot and `execSync + stdio: inherit` is really convenient here.

Reply:
Thanks for the explanation. It won't be huge matter for now, until we hit stdout limit / memory pressure with sync execution. Just curiousity q.